pyroscope_backends/pyroscope_pyspy/src/lib.rs

```diff
@@ -0,0 +1,344 @@
+use py_spy::{config::Config, sampler::Sampler};
+
+use pyroscope::{
+    backend::{Backend, Report, StackFrame, StackTrace, State},
+    error::{PyroscopeError, Result},
+};
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, Mutex,
+    },
+    thread::JoinHandle,
+};
+
+/// Pyspy Configuration
+#[derive(Debug, Clone)]
+pub struct PyspyConfig {
+    /// Process to monitor
+    pid: Option<i32>,
+    /// Sampling rate
+    sample_rate: u32,
+    /// Lock Process while sampling
+    lock_process: py_spy::config::LockingStrategy,
+    /// Profiling duration. None for infinite.
```

Comment:
nit: Some lines have trailing `.`

Reply:
@drahnr Would you say an enum (with two variants: Infinity and Duration(core::time::Duration)) is more descriptive?